### PR TITLE
Generate docs for remaining logs resources

### PR DESCRIPTION
**Pull request by nmuesch**
Generate documentation for the remaining logs resources using tfplugindocs

The top level description for logs_custom_pipeline was in the wrong place, so this PR moves that too.

### CODE REVIEW

File: docs/resources/logs_archive_order.md [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
+page_title: "datadog_logs_archive_order Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog Logs Archive API resource, which is used to manage Datadog log archives order.
 ---
 
-# datadog_logs_archive_order Resource
+# Resource `datadog_logs_archive_order`
 
-Provides a Datadog [Logs Archive API](https://docs.datadoghq.com/api/v2/logs-archives/) resource, which is used to manage Datadog log archives order.
```
Comment (zippolyte) [inline docs/resources/logs_archive_order.md:7]: Can we keep this link ?

File: docs/resources/logs_custom_pipeline.md [hunk 2]
```diff
@@ -1,275 +1,714 @@
...
+page_title: "datadog_logs_custom_pipeline Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog Logs Pipeline API resource, which is used to create and manage Datadog logs custom pipelines. Each datadog_logs_custom_pipeline resource defines a complete pipeline. The order of the pipelines is maintained in a different resource: datadog_logs_pipeline_order. When creating a new pipeline, you need to explicitly add this pipeline to the datadog_logs_pipeline_order resource to track the pipeline. Similarly, when a pipeline needs to be destroyed, remove its references from the datadog_logs_pipeline_order resource.
 ---
 
-# datadog_logs_custom_pipeline Resource
+# Resource `datadog_logs_custom_pipeline`
 
-Provides a Datadog [Logs Pipeline API](https://docs.datadoghq.com/api/v1/logs-pipelines/) resource, which is used to create and manage Datadog logs custom pipelines.
```
Comment (zippolyte) [inline docs/resources/logs_custom_pipeline.md:7]: Same here

File: docs/resources/logs_custom_pipeline.md [hunk 3]
```diff
@@ -1,275 +1,714 @@
...
+- **grok_parser** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--grok_parser))
+- **lookup_processor** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--lookup_processor))
+- **message_remapper** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--message_remapper))
+- **pipeline** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--pipeline))
+- **service_remapper** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--service_remapper))
+- **status_remapper** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--status_remapper))
+- **string_builder_processor** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--string_builder_processor))
+- **trace_id_remapper** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--trace_id_remapper))
+- **url_parser** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--url_parser))
+- **user_agent_parser** (Block List, Max: 1) (see [below for nested schema](#nestedblock--processor--user_agent_parser))
```
Comment (zippolyte) [inline docs/resources/logs_custom_pipeline.md:205]: no description ?

File: docs/resources/logs_integration_pipeline.md [hunk 4]
```diff
@@ -1,27 +1,36 @@
...
+Provides a Datadog Logs Pipeline API resource to manage the integrations.
 
-Integration pipelines are the pipelines that are automatically installed for your organization when sending the logs with specific sources. You don't need to maintain or update these types of pipelines. Keeping them as resources, however, allows you to manage the order of your pipelines by referencing them in your [datadog_logs_pipeline_order](logs_pipeline_order.html#datadog_logs_pipeline_order) resource. If you don't need the `pipeline_order` feature, this resource declaration can be omitted.
+Integration pipelines are the pipelines that are automatically installed for your organization when sending the logs with specific sources. You don't need to maintain or update these types of pipelines. Keeping them as resources, however, allows you to manage the order of your pipelines by referencing them in your datadog_logs_pipeline_order resource. If you don't need the pipeline_order feature, this resource declaration can be omitted.
```
Comment (zippolyte) [inline docs/resources/logs_integration_pipeline.md:13]:
```suggestion
Integration pipelines are the pipelines that are automatically installed for your organization when sending the logs with specific sources. You don't need to maintain or update these types of pipelines. Keeping them as resources, however, allows you to manage the order of your pipelines by referencing them in your `datadog_logs_pipeline_order` resource. If you don't need the `pipeline_order` feature, this resource declaration can be omitted.
```

File: docs/resources/logs_integration_pipeline.md [hunk 5]
```diff
@@ -1,27 +1,36 @@
 ---
-page_title: "datadog_logs_integration_pipeline"
+page_title: "datadog_logs_integration_pipeline Resource - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Provides a Datadog Logs Pipeline API resource to manage the integrations.
+  Integration pipelines are the pipelines that are automatically installed for your organization when sending the logs with specific sources. You don't need to maintain or update these types of pipelines. Keeping them as resources, however, allows you to manage the order of your pipelines by referencing them in your datadoglogspipelineorder resource. If you don't need the pipelineorder feature, this resource declaration can be omitted.
```
Comment (zippolyte) [inline docs/resources/logs_integration_pipeline.md:6]: We should maybe open an issue for the main description here doing a prerendering of the markdown. Or does it even appear on the final docs this part ??

Comment (nmuesch) [reply docs/resources/logs_integration_pipeline.md:6]: i'm not seeing this rendered on the registry site, though I could be missing it. 
As far as I can tell the `---` before and after this section are `horizontal rules` in html to break up sections.

File: docs/resources/logs_custom_pipeline.md (hunk 3 above)
Comment (nmuesch) [reply docs/resources/logs_custom_pipeline.md:205]: Good catch. All of these are missing descriptions...
The descriptions for these that I could copy from are pretty huge. WDYT of Stating the type and linking to the full docs like - https://docs.datadoghq.com/logs/processing/processors/

Comment (zippolyte) [reply docs/resources/logs_custom_pipeline.md:205]: Yeah sounds good to me

**Comment by nmuesch** [comment]
AZP issues unrelated to logs resources; merging

**Pull request merged**